middleware/auth.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+)
+
+func isAuthorized(r *http.Request) bool {
+	// TODO: Add configuration to enable this Auth
+	// This configuration should not be as command line option
+	// Because, one glusterd can be run with `--no-auth` and
+	// other glusterd with auth enabled.
+	if strings.HasPrefix(r.RemoteAddr, "127.0.0.1:") {
```

Comment:
TODO below is not for allow/deny based on IP. That will be based on Token based auth like JWT.(To be decided yet on this topic)

If IP filters to be implemented, that is entirely different feature since allowed IP list can't skip authentication.

Only `net.Parse` is not enough for comparison. I don't think constant for localhost is available in `net` package(like `LocalhostIP`), So the comparison is still based on string but with additional steps.

Example program,

    package main
     
    import (
        "net"
        "fmt"
    )
     
    func main(){
        input1 := "127.0.0.1:24007"
        input2 := "[::1]:24007"
        localhost := net.ParseIP("::1")
        host1, port1, err1 := net.SplitHostPort(input1)
        if err1 != nil{
            fmt.Println("IP split failed for input 1")
            return
        }
     
        host2, port2, err2 := net.SplitHostPort(input2)
        if err2 != nil{
            fmt.Println("IP split failed for input 2")
            return
        }   
        ip1 := net.ParseIP(host1)
        ip2 := net.ParseIP(host2)
        fmt.Println("Localhost: ", localhost,
            "\nInput1: ", input1,
            "\nInput2: ", input2,
            "\nHost 1: ", host1, " Port 1: ", port1,
            "\nHost 2: ", host2, " Port 2: ", port2,
            "\nInput 1 Equal to Localhost: ", ip1.Equal(localhost),
            "\nInput 2 Equal to Localhost: ", ip2.Equal(localhost),
        )
    }


Outputs: You can note from below output, `ip.Equal` is not suceeding for all the variant of localhost. There may be other tricks to achieve the same, but I am not aware of it.

Output When `localhost := "127.0.0.1"`

    Localhost:  127.0.0.1 
    Input1:  127.0.0.1:24007 
    Input2:  [::1]:24007 
    Host 1:  127.0.0.1  Port 1:  24007 
    Host 2:  ::1  Port 2:  24007 
    Input 1 Equal to Localhost:  true 
    Input 2 Equal to Localhost:  false

Output When `localhost := "::1"`

    Localhost:  ::1 
    Input1:  127.0.0.1:24007 
    Input2:  [::1]:24007 
    Host 1:  127.0.0.1  Port 1:  24007 
    Host 2:  ::1  Port 2:  24007 
    Input 1 Equal to Localhost:  false 
    Input 2 Equal to Localhost:  true